app/processing/statement/get-statement.js

```diff
@@ -16,7 +17,9 @@ const getStatement = async (settlementId) => {
     const address = await getAddress(sbi, transaction)
     const detailedFunding = await getDetailedFunding(calculation.calculationId, paymentRequestId, transaction)
     const scheme = await getScheme(paymentRequest)
-    const payments = await getDetailedPayments(calculation, paymentRequest, settlement)
+    const lastSettlement = await getLastSettlement(settlement, transaction)
+    const latestPayment = getLatestPayment(paymentRequest, settlement, lastSettlement)

```

Comment:
We have a (although loosely followed) standard of processing each of these 'steps' in the constructing phase were we do all or some of the following (depending on specifics) within a 'wrapper' function, i.e. `getLastSettlement` and `getLatestPayment` funcs in this case, to ensure the information we are retrieving and passing onto other functions or surfacing as this service's output is correct:
- database retrieval
- formating
- mapping
- aggregating
- validating

See `getSettlement` or `getPaymentRequest` above for reference.

We do have a tech debt on the backlog to standardise and tidy these function calls up: https://eaflood.atlassian.net/browse/PAY-2406

Would be keen to add the schema and validation here as, for example, the `getLastSettlement` returns `paymentRequestId`, `invoiceNumber`, `reference`, `settled`, `settlementDate` and `value` which I can see that not all of these keys are actually used for `lastSettlement` when it's used in `getLatestPayment` actually only `value` is so should this should be the only thing returned from `getLastSettlement`

Reply:
Makes sense, updated to match above.